tests/test_io.py

```diff
@@ -47,77 +86,50 @@ def test_load_scan_spm(load_scan: LoadScan) -> None:
 #     assert test_filters_random.pixel_to_nm_scaling == expected
 
 
+def test_load_scan_extract_jpk() -> None:
+    """Test extraction of data from loaded JPK image."""
+    assert True
+
+
 @pytest.mark.parametrize(
-    "load_scan_object, filename, image_type, image_shape, image_sum, pixel_type, pixel_to_nm_scaling, suffix",
+    "load_scan_object, length, image_shape, image_sum, filename, pixel_to_nm_scaling",
     [
-        ("load_scan", "minicircle", np.ndarray, (1024, 1024), 30695369.188316286, float, 0.4940029296875, ".spm"),
-        ("load_scan_ibw", "minicircle2", np.ndarray, (512, 512), -218091520.0, float, 1.5625, ".ibw"),
+        ("load_scan", 1, (1024, 1024), 30695369.188316286, "minicircle", 0.4940029296875),
+        ("load_scan_ibw", 1, (512, 512), -218091520.0, "minicircle2", 1.5625),
+        ("load_scan_asd", 64, (256, 256), 5958870.556640625, "minicircles_frame_40", 1.953125),
     ],
 )
 def test_load_scan_get_data(
-    load_scan_object: LoadScan,
-    filename: str,
-    image_type,
+    load_scan_object: LoadScans,
+    length: int,
     image_shape: tuple,
     image_sum: float,
-    pixel_type,
+    filename: str,
     pixel_to_nm_scaling: float,
-    suffix: str,
     request,
 ) -> None:
     """Test the LoadScan.get_data() method."""
-    scan = request.getfixturevalue(load_scan_object)
-    scan.get_data()
-    assert isinstance(scan.filename, str)
-    assert scan.filename == filename
-    assert isinstance(scan.suffix, str)
-    assert scan.suffix == suffix
-    assert isinstance(scan.image, image_type)
-    assert scan.image.shape == image_shape
-    assert scan.image.sum() == image_sum
-    assert isinstance(scan.pixel_to_nm_scaling, pixel_type)
-    assert scan.pixel_to_nm_scaling, pixel_to_nm_scaling
-
-
-def test_load_scan_ibw(load_scan_ibw: LoadScan) -> None:
-    image, px_to_nm_scaling = load_scan_ibw.load_ibw()
-    assert isinstance(image, np.ndarray)
-    assert image.shape == (512, 512)
-    assert image.sum() == -218091520.0
-    assert isinstance(px_to_nm_scaling, float)
-    assert px_to_nm_scaling == 1.5625
-
-
-# FIXME : Get this test working
-# @pytest.mark.parametrize(
-#     "px_x, px_y, re_x, re_y, expected",
-#     [
-#         (100, 100, 1e-7, 1e-7, 1),
-#         (50, 50, 1e-7, 1e-7, 2),
-#     ],
-# )
-# def test_extract_pixel_to_nm_scaling(load_scan_ibw: LoadScan, unit, x, y, expected) -> None:
-#     """Test extraction of pixels to nanometer scaling."""
-#     load_scan_ibw.load_spm()
-#     px_2_nm = load_scan_ibw._ibw_pixel_to_nm_scaling() {
-# "unit": unit, "px_x": px_x, "px_y": py_y, "re_x": re_x, "re_y": re_y
-# }
-#     assert px_2_nm == expected
-
-
-def test_load_scan_load_jpk() -> None:
-    """Test loading of JPK image."""
-    assert True
-
-
-def test_load_scan_extract_jpk() -> None:
-    """Test extraction of data from loaded JPK image."""
-    assert True
+    # temporary try statement until libasd has correct binaries
+    try:
```

Comment:
As above I don't think the `try: ... except: ...` is required as it should pass everywhere people are running.

Reply:
outdated with libasd import now available